platform/src/EducationPlatformApp.js

```diff
@@ -700,48 +704,48 @@ class EducationPlatformApp {
     }
 
     /**
-     * Poll for editor to become available. 
+     * Open a websockets connection to receive status updates on editor build. 
+     * @param {Object} socket - WebSocket object
      * @param {String} statusUrl - the url for checking the status of the editor panel.
      * @param {String} editorInstanceUrl - the editor instance's url. 
      * @param {String} editorPanelId - the id of the editor panel.
      * @param {String} editorActivityId - TODO remove as this can be found using editorPanelId to save having to specify in config.
      * @param {Panel} logPanel - the panel to log progress to.
+     * @param {Bool} editorReady - whether editor has finished building or not.
      */
-    async checkEditorReady(statusUrl, editorInstanceUrl, editorPanelId, editorActivityId, logPanel){
-
-        let response  = await fetch(statusUrl);
-
-        if (response.status == 200){ 
-            const result = await response.json();
-
-            if (result.output){
-                logPanel.setValue(result.output);
-            }
-            
-            if (result.error){
-                // Unsuccessful
-                console.log("Editor failed start.");
-                sessionStorage.removeItem(editorPanelId);
-                this.activityManager.setActivityVisibility(editorActivityId, false);
-                Metro.notify.killAll();
-                PlaygroundUtility.notification("Build Failed", result.error, "ribbed-lightAmber");
-
-            } else if (!result.editorReady){
-                await new Promise(resolve => setTimeout(resolve, 2000));
-                await this.checkEditorReady(statusUrl, editorInstanceUrl, editorPanelId, editorActivityId, logPanel);
-
-            } else {
-                // Successful 
-                console.log("Editor ready.");
+    checkEditorReady(socket, editorID, editorInstanceUrl, editorPanelId, editorActivityId, logPanel, editorReady){
+        socket = new WebSocket(this.wsUri);
+        socket.onopen = function(){
+            socket.send(editorID);
+        };
+
+        socket.onmessage = function(e){
+            var resultData = JSON.parse(e.data);
+            if (resultData.output){
+                    logPanel.setValue(resultData.output);
```

Comment:
Would be better to append output to avoid the console always jumping to the very end of the text even when nothing has actually changed.

This would require the server sending increments rather than full text. 

I would suggest that we move this into a separate issue, however, so that we can merge the core WebSocket functionality before improving its behaviour.